src/main/java/Controller/LoginController.java

```diff
@@ -18,12 +18,22 @@ public LoginController(){
      */
     public LoginController(UserManager userManager){this.uman = userManager;}
 
+    /**
+     * check if the username is a valid name for register
+     * @param name username
+     *
+     */
+    public boolean is_valid_name(String name){
+        return name.matches("^[a-zA-Z0-9]{4,12}$");
```

Comment:
It's better to put specific process in UserManager (use case layer).

Reply:
ok